airflow/providers/amazon/aws/sensors/s3_upload_session_complete.py

```diff
@@ -0,0 +1,179 @@
+        has passed and updates the state of the sensor accordingly.
+
+        :param current_objects: set of object ids in bucket during last poke.
+        :type current_objects: set[str]
+        """
+        current_num_objects = len(current_objects)
+        if current_objects > self.previous_objects:
+            # When new objects arrived, reset the inactivity_seconds
+            # and update previous_objects for the next poke.
+            self.log.info("New objects found at %s resetting last_activity_time.",
+                          os.path.join(self.bucket, self.prefix))
+            self.log.debug("New objects: %s",
+                           "\n".join(current_objects - self.previous_objects))
+            self.last_activity_time = get_time()
+            self.inactivity_seconds = 0
+            self.previous_objects = current_objects
+            return False
+
+        if self.previous_objects - current_objects:
+            # During the last poke interval objects were deleted.
+            if self.allow_delete:
+                self.previous_objects = current_objects
+                self.last_activity_time = get_time()
+                self.log.warning(
+                    """
+                    Objects were deleted during the last
+                    poke interval. Updating the file counter and
+                    resetting last_activity_time.
+                    %s
+                    """, self.previous_objects - current_objects
+                )
+                return False
+
+            raise AirflowException(
+                """
+                Illegal behavior: objects were deleted in {} between pokes.
+                """.format(os.path.join(self.bucket, self.prefix))
+            )
+
+        if self.last_activity_time:
+            self.inactivity_seconds = (get_time() - self.last_activity_time).total_seconds()
+        else:
+            # Handles the first poke where last inactivity time is None.
+            self.last_activity_time = get_time()
+            self.inactivity_seconds = 0
+
+        if self.inactivity_seconds >= self.inactivity_period:
+            path = os.path.join(self.bucket, self.prefix)
+
+            if current_num_objects >= self.min_objects:
+                self.log.info("""SUCCESS:
+                    Sensor found %s objects at %s.
+                    Waited at least %s seconds, with no new objects dropped.
+                    """, current_num_objects, path, self.inactivity_period)
+                return True
+
+            self.log.error("FAILURE: Inactivity Period passed, not enough objects found in %s", path)
+
+            return False
+        return False
+
+    def poke(self, context):
+        return self.is_bucket_updated(set(self._get_aws_hook().list_keys(self.bucket, prefix=self.prefix)))
+
+
+def get_time():
+    """
+    This is just a wrapper of datetime.datetime.now to simplify mocking in the
+    unittests.
+    """
+    return datetime.now()
```

Comment:
We're already using https://pypi.org/project/freezegun/ in the tests, so you don't need to alter the code to make it easier to test